### ISSUE TITLE
Headless service support

### ISSUE THREAD

**Issue by matteo-gsr** [rmb938/hostport-allocator#80]
This is more of a question / feature request rather than an issue itself.

I was looking at using the hostport allocator in combination with a headless service. However, I don't see any references or examples of whether this is supported or not.

Thank you.

**Comment by rmb938** [comment]
You can create a headless service that points to your pods however the hostport allocator won't do anything with it, the headless service will just act as it does normally. 

It is only made to assign ports for pod host ports.

**Comment by matteo-gsr** [comment]
understood - reason why I was asking is that I wanted to expose the allocated host port through the SRV record registered into kube dns. If that makes sense..

**Comment by rmb938** [comment]
Ah I see. What you could do is create the `HostPortClaim` and wait for it to to be `Bound`. Then do a `kubectl get hostport`, to find the actual port allocated, then create the headless service with that port. This is what I did at a previous job.

**Comment by matteo-gsr** [comment]
It's a bit manual though, ideally I would package the HostPortClaim along with the app in the same chart and then it would automatically set up the service with the allocated node port. However, I do understand this is a bit more involved and probably not easily achievable. Thank you for prompt responses. It's a great and useful project, I'll definitely use it.

**Comment by rmb938** [comment]
No problem! I can look into supporting something like that. It is a bit tricky since services act differently then pods but I'll see what I can do.

**Comment by matteo-gsr** [comment]
thanks @rmb938 - look forward to seeing what you come up with there.

**Comment by matteo-gsr** [comment]
hey @rmb938! HNY ;) - was just checking back in today after the hols as I am looking at running some tests with the controller. Did you have time to find out whether it's possible to support services at all? Thanks.

**Comment by rmb938** [comment]
Yep it should be possible. I will see if I can get something out this weekend, otherwise I'll probably be a week or two.
